v1/lists.js

```diff
@@ -0,0 +1,151 @@
+    }
+
+    /**
+     * Convert an array of values into the format expected by the MediaWiki API.
+     * @param {!Array} list A list containing strings and numbers.
+     * @return {!String}
+     */
+    flattenMultivalue(list) {
+        return list.join('|');
+    }
+
+    /**
+     * Handle the /list/{id}/entries endpoint (get entries of a list).
+     * @param {!HyperSwitch} hyper
+     * @param {!Object} req The request object as provided by HyperSwitch.
+     * @return {!Promise<Object>} A response promise.
+     */
+    getListEntries(hyper, req) {
+        return hyper.post({
+            uri: new URI([req.params.domain, 'sys', 'action', 'rawquery']),
+            body: {
+                action: 'query',
+                list: 'readinglistentries',
+                rlelists: req.params.id,
+                rlelimit: 'max',
+                continue: this.unflattenContinuation(req.query.next).continue,
+                rlecontinue: this.unflattenContinuation(req.query.next).rlecontinue,
+            },
+        })
+        .then((res) => {
+            const entries = res.body.query.readinglistentries;
+            const next = this.flattenContinuation(res.body.continue);
+
+            return this.hydrateSummaries(entries, hyper, req)
+            .then(entries => ({
+                status: 200,
+                headers: {
+                    'content-type': 'application/json; charset=utf-8;'
+                        + 'profile="https://www.mediawiki.org/wiki/Specs/Lists/0.1"',
+                    'cache-control': 'no-cache',
+                },
+                body: {
+                    entries,
+                    next,
+                }
+            }));
+        });
+    }
+
+    /**
+     * Add data from the summary endpoint to an array of list entries.
+     * @param {!Array} entries An array of objects with two properties:
+     *   - project: the domain of the wiki
+     *   - title: the title of the page
+     * @param {!HyperSwitch} hyper
+     * @param {!Object} req The request object as provided by HyperSwitch.
+     * @return {!Promise<Array>} The objects, enriched with summaries.
+     */
+    hydrateSummaries(entries, hyper, req) {
+        return mwUtil.getSiteMatrix(hyper, req).then((sitematrix) => {
+            return P.map(entries, (entry) => {
+                const site = sitematrix[entry.project];
+                if (site && !site.private) {
+                    const uri = `https://${entry.project}/api/rest_v1/page/summary/`
+                        + `${encodeURIComponent(entry.title)}`;
+                    return mwUtil.fetchSummary(hyper, uri)
+                        .then((res) => {
+                            if (res) {
+                                entry.summary = res;
+                            }
+                            return entry;
+                        });
+                }
+                return entry;
+            }, { concurrency: this.options.summaryConcurrency });
```

Comment:
I still think duplicating code that mostly exists in the `hydrateSummaries` is not great and I would personally prefer if we mangle the result here to use the `$merge` and filter out private sites and then reuse the `hydrate` thingy. Just like [here](https://github.com/wikimedia/restbase/blob/master/v1/related.js#L31-L48)

Adding support for concurrency to the generic `hydrateResponse` is a great idea anyway and I will do that regardless of whether we decide to add it now or not although 10 is a bit too low I think. We can easily do 100.

Anyways feel free to disregard this comment, I don't have a very strong opinion here.

Reply:
@tgr we can change that behaviour. Just need to ask @berndsi if that will affect mobile clients that's using feeds. Or we can put that under a parameter flag like`removeFailed = true|false`